### ISSUE TITLE
Html escaping escapes characters inside <script> tags if string is imported

### ISSUE THREAD

**Issue by yoroshikun** [brillout/vite-plugin-ssr#83]
Currently, the HTML escaper escapes characters insider the <script> tag, this should not be the case

eg.
```
const string = `const theme = sessionStorage.getItem('theme');`
<script>
${string}
</script>
```
becomes incorrect syntax
```
<script>
const theme = sessionStorage.getItem(&#039;theme&#039;);
</script>
```

Seems to affect all "imported" strings

**Comment by brillout** [comment]
Yes this is by design to avoid malicious code injection.

You can use `html.dangerouslySetHtml()` to avoid escaping.

Maybe we should rename `dangerouslySetHtml` as it doesn't have to be HTML. The only thing `dangerouslySetHtml` does is to skip escaping.

**Comment by yoroshikun** [comment]
Oh, that completely skipped my mind, that should definitely fix the issue!

**Comment by brillout** [comment]
Yea but I feel like we should rename `dangerouslySetHtml`.

Maybe `dangerouslySkipEscaping` instead?

**Comment by yoroshikun** [comment]
I am not too bothered with the rename or not as from react a similar feature is name [dangerouslySetInnerHtml](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml)
I was not aware and putting two and two together for using already available `html.dangerouslySetHtml()`

P.S
As expected using `html.dangerouslySetHtml()` worked as intended to fix my issue.

**Comment by brillout** [comment]
Rename is released in `0.1.0-beta.48`.
